File: bionic-js-tool/testing-code/guest/libs/FuelType.js
class FuelType {

    // @bionic FuelType
    static get Electricity() {
        return new FuelType('electricity', 0.2)
    }

    // @bionic FuelType
    static get NaturalGas() {
        return new FuelType('natural gas', 0.4)
    }

    // @bionic FuelType
    static get Diesel() {
        return new FuelType('diesel', 0.6)
    }

    // @bionic FuelType
    static get Petrol() {
        return new FuelType('petrol', 0.8)
    }

    // @bionic FuelType
    static get Kerosene() {
        return new FuelType('kerosene', 1.0)
    }
    
    constructor(name, cost) {
        this.name = name
        this.cost = cost
    }

    // @bionic get name String
    // @bionic get cost Float
}

module.exports = {FuelType}

File: bionic-js-tool/testing-code/java/project-with-host-files/HostProject/src/main/java/test/project/host/libs/FuelType.java
package test.project.libs;

import jjbridge.api.runtime.JSReference;
import bionic.js.Bjs;
import bionic.js.BjsTypeInfo;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.BjsObject;

@BjsTypeInfo.BjsLocation(project = "BeautifulVehicles", module = "FuelType")
public class FuelType extends BjsObject {
    
    public static FuelType Electricity() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Electricity"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType NaturalGas() {
        return bjs.getObj(bjs.getProperty(bjsClass, "NaturalGas"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Diesel() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Diesel"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Petrol() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Petrol"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Kerosene() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Kerosene"), FuelType.bjsFactory, FuelType.class);
    }
    
    public String name() {
        return bjs.getString(bjsGetProperty("name"));
    }
    
    public Double cost() {
        return bjs.getDouble(bjsGetProperty("cost"));
    }
    
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(FuelType.class).bjsClass();
    public static final Bjs bjs = BjsObjectTypeInfo.get(FuelType.class).bjsLocator.get();
    public static final Bjs.JSReferenceConverter<FuelType> bjsFactory = FuelType::new;
}